From 4fcbdb83790ddbc2088b3d56ab9a7b677b735ded Mon Sep 17 00:00:00 2001
From: Pavan Kumar S <pavan.kumar.s@intel.com>
Date: Mon, 21 Oct 2013 16:39:45 +0530
Subject: [PORT FROM MAIN] RECOVERY/minui: Correct surface format initialization

BZ: 110674

During gr_init, surface-format is getting initialized to
GGL_PIXEL_FORMAT_L_8, which is 1 bytes per pixel.
Since the pixel stride is 4 bytes in most cases except PNG_COLOR_TYPE_GRAY,
the surface format should also be 4 bytes.

Orig-Change-Id: If839cd677ddbd2683f1b12a590f01ddc84ac8d6b
Category: device enablement
Domain: PUPDR-Recovery
Origin: internal
Upstream-Candidate: no

Change-Id: I2ef588a0688ba3d4e7d7fb8dabc64c9a697dc76f
Signed-off-by: Pavan Kumar S <pavan.kumar.s@intel.com>
---
 minui/resources.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/minui/resources.c b/minui/resources.c
index 72f39fb..df446c7 100644
--- a/minui/resources.c
+++ b/minui/resources.c
@@ -122,7 +122,7 @@ int res_create_surface(const char* name, gr_surface* pSurface) {
     surface->stride = width; /* Yes, pixels, not bytes */
     surface->data = pData;
     surface->format = (channels == 3) ? GGL_PIXEL_FORMAT_RGBX_8888 :
-        ((color_type == PNG_COLOR_TYPE_PALETTE ? GGL_PIXEL_FORMAT_RGBA_8888 : GGL_PIXEL_FORMAT_L_8));
+        GGL_PIXEL_FORMAT_RGBA_8888;
 
     int alpha = 0;
     if (color_type == PNG_COLOR_TYPE_PALETTE) {
-- 
1.7.4.1

